cli/src/authorization/commands.py

```diff
@@ -0,0 +1,111 @@
+    assert len(security_groups) == 1
+    ip_permissions = security_groups[0]['IpPermissions']
+    ip_perm = next(
+        filter(lambda rule: rule['FromPort'] == port, ip_permissions))
+    return any(filter(lambda x: x['CidrIp'] == cidr_ip, ip_perm['IpRanges']))
+
+
+def configure_security_groups_ingress(zone, *, owner=None, revoke=False):
+    cidr_ip = f'{get_external_ip()}/32'
+    region_name = get_aws_region_name(zone)
+    ec2 = boto3.client('ec2', region_name=region_name)
+
+    for security_group in SECURITY_GROUPS:
+        group_names = [x % dict(zone=zone)
+                       for x in security_group['group_name_pattern']]
+        res = ec2.describe_security_groups(
+            Filters=[dict(Name='group-name', Values=group_names)])
+        for group in res['SecurityGroups']:
+            group_id = group['GroupId']
+            group_name = f'{group["GroupName"]} ({group_id})'
+            description = security_group['description']
+            if owner:
+                description += f' for {owner}'
+            for port in security_group['ports_to_open']:
+                name = f'{description}: CIDR IP {cidr_ip} for port {port} in {group_name}'
+                is_defined = is_cidr_ip_defined_in_security_group_ingress(
+                    ec2, group_id, port, cidr_ip)
+                if revoke ^ is_defined:
+                    defined_status = 'already defined' if is_defined else 'not defined'
+                    click.echo(f'{name}: {defined_status} - skipping')
+                    continue
+
+                if revoke:
+                    func = ec2.revoke_security_group_ingress
+                    ip_ranges = [{'CidrIp': cidr_ip}]
+                else:
+                    func = ec2.authorize_security_group_ingress
+                    ip_ranges = [
+                        {'CidrIp': cidr_ip, 'Description': description}]
+
+                func(
+                    GroupId=group_id,
+                    IpPermissions=[
+                        {
+                            'IpRanges': ip_ranges,
+                            'FromPort': port,
+                            'IpProtocol': 'TCP',
+                            'ToPort': port,
+                        }
+                    ]
+                )
+                action = 'removed' if revoke else 'defined'
+                click.echo(f'{name}: {action}')
+
+
+@click.group()
+def authorization():
+    """Control AWS security groups"""
+    pass
+
+
+@authorization.command()
+def my_external_ip():
+    """Get current external IP"""
+    click.echo(get_external_ip())
+
+
+@authorization.command()
+@click.option('-z', '--zone', type=click.Choice(ZONES, case_sensitive=False), default=DEFAULT_ZONE)
+@click.option('-o', '--owner', required=True, default=os.getenv('USER'), help='Owner of the added access rules')
+def authorize_my_ip(zone, owner):
+    """Authorize security groups with current external IP address"""
```

Comment:
I assume i should already be logged in into the cli right?

Reply:
Yes, you should have your AWS CLI credentials set. Otherwise, the script will fail.
I don't know how I can automate this. Maybe we can add an interactive stage to the dev-env installation script, where it will ask you for this info. Let's open a ticket for it, and do it separately: https://github.com/meirim-org/meirim/issues/670.

Reply:
> I assume i should already be logged in into the cli right?

Clarification - you are not supposed to be logged into the AWS CLI for the `meirim` CLI tool to work.
Some commands, like this one, do require AWS CLI to be configured. I added a config command to configure the AWS CLI tool:

```
> meirim config aws-cli-setup
```